gui/src/app/SPAnalysis/SPAnalysisDataModel.ts

```diff
@@ -42,17 +44,26 @@ export const initialDataModel: SPAnalysisDataModel = {
     samplingOpts: defaultSamplingOpts
 }
 
-export const serializeAnalysis = (data: SPAnalysisDataModel): string => {
-    const intermediary = {
-        ...data, ephemera: undefined }
-    return JSON.stringify(intermediary)
+export const persistStateToEphemera = (data: SPAnalysisDataModel): SPAnalysisDataModel => {
+    const newEphemera = { ...data.ephemera }
+    getStringKnownFileKeys().forEach(k => newEphemera[k] = data[k])
+    return {
+        ...data,
+        ephemera: newEphemera
+    }
 }
 
-export const deserializeAnalysis = (serialized: string): SPAnalysisDataModel => {
-    const intermediary = JSON.parse(serialized)
-    // Not sure if this is strictly necessary
-    intermediary.ephemera = {}
-    const stringFileKeys = Object.values(SPAnalysisKnownFiles).filter((v) => isNaN(Number(v)));
-    stringFileKeys.forEach((k) => intermediary.ephemera[k] = intermediary[k]);
-    return intermediary as SPAnalysisDataModel
+export const getStringKnownFileKeys = () => Object.values(SPAnalysisKnownFiles).filter((v) => isNaN(Number(v)));
+
+export const modelHasUnsavedChanges = (data: SPAnalysisDataModel): boolean => {
+    const stringFileKeys = getStringKnownFileKeys()
+    return stringFileKeys.every((k) => data[k] !== data.ephemera[k])
```

Comment:
Should this be `some` instead of `every`?

Reply:
It sure should...